presto-main/src/main/java/io/prestosql/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -240,7 +240,7 @@ public PlanWithProperties visitMarkDistinct(MarkDistinctNode node, HashComputati
         {
             // skip hash symbol generation for single bigint
             if (canSkipHashGeneration(node.getDistinctSymbols())) {
-                return planSimpleNodeWithProperties(node, parentPreference);
+                return planSimpleNodeWithProperties(node, parentPreference, false);
```

Comment:
What is the impact of this change.  I would assume in the case of mark distinct of a bigint, you would not need the extra hashes?

Reply:
For queries as described here in prestodb/presto#8024 we might need extra hashes for the subsequent exchange nodes and since we have pruned extra hashes it gets recomputed once again when required at later stages.